### ISSUE TITLE
VSCode Plugin Shows Cannot Resolve Module Error for Standard Library Modules

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/ballerina-lang#33765]
**Description:**

When I open a ballerina project (the Ballerina GraphQL module repo) on VSCode, I get `Cannot Resolve Module <module name>` error for standard library modules. See the screenshot:
![Screenshot 2021-11-11 at 14 46 20](https://user-images.githubusercontent.com/40016057/141271230-8837e39b-dfe1-408a-8cf5-340b43f9c7e5.png)

I have set the Ballerina home path as `/Library/Ballerina/distributions/ballerina-slbeta3`, (Tried removing this as well without luck). It seems like the Ballerina distribution is picked correctly. Check the following screenshot:
![Screenshot 2021-11-11 at 14 47 27](https://user-images.githubusercontent.com/40016057/141271613-f3180b75-ca43-4738-b152-03dfb08f9179.png)

Following is the console log from the ballerina:
```shell
Active the Ballerina VS Code extension.
ballerina command _stderr: /bin/sh: ballerina: command not found
ballerina command err: Error: Command failed: ballerina version
/bin/sh: ballerina: command not found
bal command stdout: Ballerina Swan Lake Beta 3
Language specification 2021R1
Update Tool 1.3.5
'bal' executor is picked up by the plugin.
Ballerina version output: Ballerina Swan Lake Beta 3
Language specification 2021R1
Update Tool 1.3.5
Plugin version: 2.1.1
Ballerina version: Swan Lake Beta 3
Using Ballerina CLI command 'bal' for Language server.
LS offline source compilation set to true
[Info  - 2:43:23 PM] Operation 'text/didOpen' {project: 'file:///Users/thisaru/Projects/Ballerina/module-ballerina-graphql/ballerina/' kind: 'build_project'} created
[Info  - 2:43:24 PM] Operation 'text/didOpen' {fileUri: 'file:///Users/thisaru/Projects/Ballerina/module-ballerina-graphql/ballerina/tests/utils.bal'} opened
[Error - 2:43:27 PM] Operation 'text/codeAction' failed! {uri: '/Users/thisaru/Projects/Ballerina/module-ballerina-graphql/ballerina/tests/utils.bal', [19:21]- [19:21], error: 'java.util.ConcurrentModificationException'}
java.util.ConcurrentModificationException
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1661)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.ballerinalang.langserver.codeaction.CodeActionProvidersHolder.getActiveDiagnosticsBasedProviders(CodeActionProvidersHolder.java:110)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$5(CodeActionRouter.java:97)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:95)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:293)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by nadeeshaan** [comment]
Did you try removing the `target` folder from the project or the `Dependencies.toml`

**Comment by sameerajayasoma** [comment]
This is not released to project API IMO? Let's discuss this during the call. :)

**Comment by azinneera** [comment]
@nadeeshaan AFAIR, this started coming after introducing a service to render diagram and @prabushi fixed it to access the project instance from the project map that LS maintains so that the compilation is done sequentially.

**Comment by prabushi** [comment]
This was resolved for the `BallerinaPackageService` in https://github.com/ballerina-platform/ballerina-lang/issues/32820
It seems the mentioned has a different stack trace. Is there a way to replicate this issue?

**Comment by nadeeshaan** [comment]
@prabushi I think the provided stack trace is not really associated with this particular issue raised. IIRC, this dependency issue was addressed with #32280.

**Comment by azinneera** [comment]
This does not seem to be related to the Project API. Anyway, since the issue is said to be fixed, we are closing the issue. @ThisaruGuruge please reopen if you get the issue again.
